### ISSUE TITLE
Path commands maybe shouldn't strip extensions

### ISSUE THREAD

**Issue by oefd** [XaF/omni#88]
Specifically talking about [this](https://github.com/XaF/omni/blob/main/src/internal/commands/frompath.rs#L58-L63)

It's possible that 2 or more commands will end up with identical names due to having different extensions (or "extensions" which are really just incidentally a dot being in the name). The case that made me scratch my head and go looking for what happened was this

```
$ omni
omni - omnipotent tool (v0.0.13-24-g60a2ab6-dirty)
...
Uncategorized
  python, python3, python3
```

In this case the real files picked up as commands were `python`, `python3`, and `python3.11` which are all in a standard python virtual environment.

Possible ways to address this:

* Don't worry about it, it's kind of a niche thing
* Don't truncate the extension, people can rename the file and drop the extension themselves if they like
* Only truncate some common shell/utility extensions (`sh`, `bash`, `zsh`, `py`, `js`, etc)
* If any duplicate path commands are found remove both from the list and emit a warning

**Comment by XaF** [comment]
Interesting!
The second `python3` shouldn't  show up in the list and have been ignored, this is a bug!
As stated in the documentation:
> Path commands are also loaded in the order in which they are defined in the omnipath, and alphabetically after that. This means that if a path command has the same name as a [builtin command](https://omnicli.dev/reference/builtin-commands), [configuration command](https://omnicli.dev/reference/custom-commands/configuration), or another path command loaded from a path that has priority in the omnipath, it will be ignored.

The choice of not emitting a warning is because it would then get very noisy, or require caching that we warned the user about it. So instead, it is just being ignored (like the regular "PATH" works).

**Comment by XaF** [comment]
Seems to not dedup when adding aliases:

```
Uncategorized
  cmd1, cmd1
```

with:

```
$ ls -alsh cmd1*
0 lrwxr-xr-x 1 xaf xaf 6 Aug  8 17:22 cmd1 -> cmd1.1
0 -rwxr-xr-x 1 xaf xaf 0 Aug  8 17:21 cmd1.1
```

**Comment by XaF** [comment]
Fixed in 9bbf1324503c825cddd36000a06de9af6414de6f

**Comment by XaF** [comment]
Also made me discover another bug fixed in 7bced93e91907af59a4eee1277b1b1d9a954d2dc
